lib/Model.js

```diff
@@ -542,363 +554,374 @@ Model.update = function(NewModel, key, update, options, next) {
+        }
       }
     }
-  }
 
-  operations.getUpdateExpression(updateReq);
+    operations.getUpdateExpression(updateReq);
 
-  // AWS doesn't allow empty expressions or attribute collections
-  if(!updateReq.UpdateExpression) {
-    delete updateReq.UpdateExpression;
-  }
-  if(!Object.keys(updateReq.ExpressionAttributeNames).length) {
-    delete updateReq.ExpressionAttributeNames;
-  }
-  if(!Object.keys(updateReq.ExpressionAttributeValues).length) {
-    delete updateReq.ExpressionAttributeValues;
-  }
+    // AWS doesn't allow empty expressions or attribute collections
+    if(!updateReq.UpdateExpression) {
+      delete updateReq.UpdateExpression;
+    }
+    if(!Object.keys(updateReq.ExpressionAttributeNames).length) {
+      delete updateReq.ExpressionAttributeNames;
+    }
+    if(!Object.keys(updateReq.ExpressionAttributeValues).length) {
+      delete updateReq.ExpressionAttributeValues;
+    }
 
-  const newModel$ = NewModel.$__;
+    const newModel$ = NewModel.$__;
 
-  function updateItem () {
-    debug('updateItem', updateReq);
-    newModel$.base.ddb().updateItem(updateReq, function(err, data) {
-      if(err) {
-        debug('Error returned by updateItem', err);
-        return deferred.reject(err);
-      }
-      debug('updateItem response', data);
-
-      if(!Object.keys(data).length) {
-        return deferred.resolve();
-      }
+    function updateItem () {
+      debug('updateItem', updateReq);
+      newModel$.base.ddb().updateItem(updateReq, function(err, data) {
+        if(err) {
+          debug('Error returned by updateItem', err);
+          return reject(err);
+        }
+        debug('updateItem response', data);
 
-      const model = new NewModel();
-      model.$__.isNew = false;
-      try {
-        schema.parseDynamo(model, data.Attributes);
-      } catch(e){
-        debug('cannot parse data', data);
-        return deferred.reject(e);
-      }
+        if(!Object.keys(data).length) {
+          return resolve();
+        }
 
-      debug('updateItem parsed model', model);
+        const model = new NewModel();
+        model.$__.isNew = false;
+        try {
+          schema.parseDynamo(model, data.Attributes);
+        } catch(e){
+          debug('cannot parse data', data);
+          return reject(e);
+        }
 
-      deferred.resolve(model);
-    });
-  }
+        debug('updateItem parsed model', model);
 
-  if(newModel$.options.waitForActive) {
-    newModel$.table.waitForActive().then(updateItem).catch(deferred.reject);
-  } else {
-    updateItem();
-  }
+        resolve(model);
+      });
+    }
 
-  return deferred.promise.nodeify(next);
+    if(newModel$.options.waitForActive) {
+      newModel$.table.waitForActive().then(updateItem).catch(reject);
+    } else {
+      updateItem();
+    }
+  })
 };
 
-Model.delete = function(NewModel, key, options, next) {
+Model.delete = async function (NewModel, key, options, next) {
+
+  options = options || {};
```

Comment:
Why did this section get added? At first glance it looks good, but it wasn't in the old version. Was that a bug in the old version or?

Reply:
I was trying to keep it consistent with the other function apis. If you want though, I can remove it from this change.